libs/execution_context/ebpf_program.c

```diff
@@ -528,38 +491,45 @@ ebpf_program_initialize(_Inout_ ebpf_program_t* program, _In_ const ebpf_program
             program_parameters->program_info_hash_length);
     }
 
-    program->parameters = *program_parameters;
+    local_program->parameters = *program_parameters;
 
-    program->parameters.program_name = local_program_name;
+    local_program->parameters.program_name = local_program_name;
     local_program_name.value = NULL;
-    program->parameters.section_name = local_section_name;
+    local_program->parameters.section_name = local_section_name;
     local_section_name.value = NULL;
-    program->parameters.file_name = local_file_name;
+    local_program->parameters.file_name = local_file_name;
     local_file_name.value = NULL;
 
-    program->parameters.code_type = EBPF_CODE_NONE;
-    program->parameters.program_info_hash = local_program_info_hash;
+    local_program->parameters.code_type = EBPF_CODE_NONE;
+    local_program->parameters.program_info_hash = local_program_info_hash;
     local_program_info_hash = NULL;
 
-    ebpf_lock_unlock(&program->lock, state);
-    lock_held = false;
+    retval = _ebpf_program_load_providers(local_program);
+    if (retval != EBPF_SUCCESS) {
+        goto Done;
+    }
 
-    return_value = _ebpf_program_load_providers(program);
-    if (return_value != EBPF_SUCCESS) {
+    retval = ebpf_object_initialize(
```

Comment:
same here: adding a comment that this needs to be the last step.

Reply:
Fixed